### ISSUE TITLE
Research - Integration with APTS to use the Test AS AN Agent approach to do Negative testing

### ISSUE THREAD

**Issue by nodlesh** [openwallet-foundation/owl-agent-test-harness#99]
The "backchannel" is a Python API - you copy this file into your repo and edit it to support your agent:

https://github.com/ianco/aries-protocol-test-suite/blob/master/apts/src/aut.py

(Actually you copy the whole repo into your agent repo)

It's possible (although probably not trivial) to map the APTS "backchannel API" to AATH backchannel (theirs is a sub-set) and create a version of their `aut.py` that calls an AATH backchannel using http calls ... This may help in taking advantage of the Test IS AN Agent approach to do negative and exception testing, and maybe to run AATH tests through the APTS.

More productive would be to rewrite their `aut.py` to use an http interface rather than native Python calls, and change the APTS to call their backchannel using http, in which case we could converge the two and use the same backchannel for both test suites.

In summary, APTS:
- you have to copy the APTS repo into your agent repo and edit it
- not sure how you manage updates to the test coverage after you'e already copied it over
- supports only Python (?) since it uses a native Python API for the backchannel
